### ISSUE TITLE
ESP8266 Wifi Library refactored, build broken

### ISSUE THREAD

**Issue by sle118** [mysensors/MySensors#625]
Trying to move up to MySensors 2.0, I am getting compile errors on the ESP8266.  The issue seems to come from a refactoring of the ESP8266 wifi library: https://github.com/esp8266/Arduino/commit/8db4dcea426b2a1b7d1cca6d5bf4c351284f2240

Issue was reported in the forum
https://forum.mysensors.org/topic/3182/how-to-standalone-node-on-esp8266-using-wifi-only-sketch-problem

errors generated are similar to this
```
WiFiClient.h: In instantiation of 'size_t WiFiClient::write(T&, size_t) [with T = char*; size_t = unsigned int]':
C:\Users\braml\Documents\Arduino\libraries\MySensors/core/MyGatewayTransportEthernet.cpp:164:52: required from here
WiFiClient.h:113:36: error: request for member 'available' in 'source', which is of non-class type 'char*'
size_t left = source.available();
^
WiFiClient.h:117:5: error: request for member 'read' in 'source', which is of non-class type 'char*'
source.read(buffer.get(), will_send);
^
```

**Comment by sle118** [comment]
Can anyone comment on which version of the ESP8266 library is supported?  Looks like there were significant changes done on the library over the past few months. Whatever is currently available right now from the Arduino board manager doesn't work.

**Comment by sle118** [comment]
for the record, compiling the gateway for version 2.0+ of MySensors as of October 2016 requires using the git version of the ESP8266 Arduino library as described here
https://github.com/esp8266/Arduino#using-git-version

this will include changes that are scheduled to be released for the ESP8266 Arduino library V 2.4
